algorithm1/openapi.json

```diff
@@ -0,0 +1,245 @@
+{
+  "openapi" : "3.0.0",
+  "info" : {
+    "version" : "2.0",
+    "title" : "algorithm-1",
+    "description" : "This is the Algorithm-1 API endpoint specs for SENG499 Capstone Company 3 & 4."
+  },
+  "paths" : {
+    "/schedule" : {
+      "post" : {
+        "summary" : "Generate schedule(s)",
+        "description" : "Takes 3 inputs and returns 1 to 3 semesters for Fall, Spring, and Summer terms accordingly. Would return semester schedules for filled fields. Therefore, if fall is the only field then the other will be empty.",
+        "responses" : {
+          "200" : {
+            "description" : "Schedule(s) generated successfully and returned.",
+            "content" : {
+              "application/json" : {
+                "schema" : {
+                  "type" : "object",
+                  "items" : {
+                    "$ref" : "#/components/schemas/Schedule"
+                  },
+                  "example" : {
+                    "fallCourses" : [ ],
+                    "SpringCourses" : [ ],
+                    "SummerCourses" : [ ]
+                  }
+                }
+              }
+            }
+          },
+          "400" : {
+            "description" : "Error generating schedule(s)."
+          }
+        },
+        "requestBody" : {
+          "description" : "This is the request body format.",
+          "required" : true,
+          "content" : {
+            "application/json" : {
+              "schema" : {
+                "type" : "object",
+                "properties" : {
+                  "hardScheduled" : {
```

Comment:
Could a description be added to this? I'm not exactly sure what it means here.

Reply:
Basically its courses that are already scheduled at a specific time before the algorithm even runs. This is in the description. This would be courses like CHEM, PHYS, ... or courses that Rich want at a specific time.